src/main/java/lotto/domain/LottoMachine.java

```diff
@@ -9,19 +9,37 @@ public class LottoMachine {
     public static final int PRICE_OF_LOTTO = 1_000;
 
     public List<LottoNumbers> buy (long money) {
-        validate(money);
+        return this.buy(money, new ArrayList<>());
+    }
+
+    public List<LottoNumbers> buy (long money, List<String> manualLottoNumberInputs) {
+        validate(money, manualLottoNumberInputs);
 
         List<LottoNumbers> lottos = new ArrayList<>();
+
+        money = money - (PRICE_OF_LOTTO * manualLottoNumberInputs.size());
+        for(String manualLottoNumberInput : manualLottoNumberInputs) {
+            lottos.add(LottoNumbers.toLottoNumber(manualLottoNumberInput));
+        }
+
         IntStream.range(0, numberOfLottoTickets(money))
                 .forEach(i -> lottos.add(new LottoNumbers()));
         return lottos;
     }
```

Comment:
```suggestion
    public List<LottoNumbers> buy(long money, List<String> manualLottoNumberInputs) {
        validate(money, manualLottoNumberInputs);

        List<LottoNumbers> lottos = new ArrayList<>();
        lottos.addAll(generateManualLottos(manualLottoNumberInputs));
        lottos.addAll(generateAutoLottos(calculateRemainingMoney(money, manualLottoNumberInputs)));

        return lottos;
    }
```

이런 방식으로 메서드를 좀더 세분화하여 하는 일을 이름지어 코드를 명확하게 작성해보는것도 좋을것 같아요!

Reply:
💯 저도 좋아하는 방식입니다. 좋은 조언 감사합니다.